src/replication.c

```diff
@@ -951,7 +951,7 @@ void syncCommand(client *c) {
         }
 
         if (!strcasecmp(c->argv[1]->ptr,server.replid)) {
-            replicationUnsetMaster();
+            clusterPromoteSelfToMaster();
```

Comment:
@oranagra I do not understand your comment. This is a rename for the sake of clarity and to put things more into cluster teminology.

Reply:
This is also only for standalone, the cluster coordinated failover works over the clusterbus, maybe we should even add a check here for not cluster mode?

Reply:
just spoke to @madolson about this. What we said was that we want to add an "if cluster mode" to this code to make it more self contained and more explicit (even though this code can never be reached when not in cluster mode).

Reply:
We can't panic unfortunately because we don't want to allow some elevated admin user to cause crashes easily. Perhaps clusterPromoteSelfToMaster should allow throwing an error.

Reply:
and then what is done with the error when it is caught?

Reply:
i don't consider PSYNC FAILOVER a feature that's cluster specific, so this code is not cluster specific.
the suggestion to make it an `if-else` does improve the situation a bit, but i wonder if we can't just run both
i.e. do we see a case where in the cluster mode we won't need to promote the local instance?

Reply:
@oranagra, we'll talk but if I understand you correctly, in the current cluster implementation there is no cluster bus support for implementing PSYNC FAILOVER. This change makes it possible for *other* cluster implementations to support PSYNC FAILOVER.